packages/language/src/language-server/types.ts

```diff
@@ -109,14 +113,90 @@ export function severityToLsp(severity: Severity): lsp.DiagnosticSeverity {
 
 export interface Diagnostic {
   severity: Severity;
-  uri: string;
-  range: Range;
+  uri?: string;
+  range?: Range;
   message: string;
   code?: string;
   data?: any;
   source?: string;
 }
 
+export function diagnosticFromCode(
```

Comment:
Nice function!

Reply:
But I guess the parameters are without names for autocompletion, right?
Is there a chance to get those? Like: `diagnosticFromCode(PLICodes.Error.IBM3581I, argTokens.nameToken)(argTokens.nameToken.image)`? Might look ugly, and maybe it goes also without second parentheses. I think you could infer the message type with a given IBM code. Have you tried it out?

Reply:
Or are those incomplete messages the INVALID ones you were talking about?

Reply:
> I think you could infer the message type with a given IBM code. Have you tried it out?

We could, but unfortunately, all the type information about the exact message is erased right now, since it is always casted to `as ParametricPLICode`.